clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/InternalLocalBootstrappingVeniceChangelogConsumer.java

```diff
@@ -308,6 +312,10 @@ private Collection<PubSubMessage<K, ChangeEvent<V>, VeniceChangeCoordinate>> pol
       currentPartitionState.currentPubSubPosition = record.getOffset();
```

Comment:
So I 'thought' this is where we figured the bug was?  Doing the checkpointing just based off of currentPubSubPosition from the last record consumed isn't quite the right thing to do since the poll method is going to screen out control messages.

Looking at the code in kafkaConsumer I could have swore we had a method for asking the consumer the high offset, but it looks like I am mistaken....  I'll put up a change for this.

Reply:
Yes, if there is any control message got filtered out we'll get stuck here. I didn't change it in this PR as currently this will fail the EI only as it doesn't have frequent updates.